From ec154451a307c3fe549c6ea17ca9ecd650f04180 Mon Sep 17 00:00:00 2001
From: TungDang <tung.dang.jy@rvc.renesas.com>
Date: Sun, 21 Aug 2016 10:08:12 +0700
Subject: [PATCH 8/9] waylandsink: Change the color format mapping for XBGR8888
 in wl_kms

This change fixes the format interpretation difference between
GStreamer and the wayland server side in wl_kms.
The mapping for BGRX8888 is also changed to keep symmetry of
the interpretation.

Porting from a patch in Linux 3.10 environment made by Kazunori Kobayashi <kkobayas@igel.co.jp>

Signed-off-by: TungDang <tung.dang.jy@rvc.renesas.com>
---
 ext/wayland/wlvideoformat.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/ext/wayland/wlvideoformat.c b/ext/wayland/wlvideoformat.c
index b1260bd..8063c26 100644
--- a/ext/wayland/wlvideoformat.c
+++ b/ext/wayland/wlvideoformat.c
@@ -87,18 +87,18 @@ static const wl_VideoFormat formats[] = {
 #if G_BYTE_ORDER == G_BIG_ENDIAN
   {WL_KMS_FORMAT_XRGB8888, GST_VIDEO_FORMAT_xRGB},
   {WL_KMS_FORMAT_ARGB8888, GST_VIDEO_FORMAT_ARGB},
-  {WL_KMS_FORMAT_XBGR8888, GST_VIDEO_FORMAT_xBGR},
+  {WL_KMS_FORMAT_XBGR8888, GST_VIDEO_FORMAT_BGRx},
   {WL_KMS_FORMAT_RGBX8888, GST_VIDEO_FORMAT_RGBx},
-  {WL_KMS_FORMAT_BGRX8888, GST_VIDEO_FORMAT_BGRx},
+  {WL_KMS_FORMAT_BGRX8888, GST_VIDEO_FORMAT_xBGR},
   {WL_KMS_FORMAT_ABGR8888, GST_VIDEO_FORMAT_ABGR},
   {WL_KMS_FORMAT_RGBA8888, GST_VIDEO_FORMAT_RGBA},
   {WL_KMS_FORMAT_BGRA8888, GST_VIDEO_FORMAT_BGRA},
 #else
   {WL_KMS_FORMAT_XRGB8888, GST_VIDEO_FORMAT_BGRx},
   {WL_KMS_FORMAT_ARGB8888, GST_VIDEO_FORMAT_BGRA},
-  {WL_KMS_FORMAT_XBGR8888, GST_VIDEO_FORMAT_RGBx},
+  {WL_KMS_FORMAT_XBGR8888, GST_VIDEO_FORMAT_xRGB},
   {WL_KMS_FORMAT_RGBX8888, GST_VIDEO_FORMAT_xBGR},
-  {WL_KMS_FORMAT_BGRX8888, GST_VIDEO_FORMAT_xRGB},
+  {WL_KMS_FORMAT_BGRX8888, GST_VIDEO_FORMAT_RGBx},
   {WL_KMS_FORMAT_ABGR8888, GST_VIDEO_FORMAT_RGBA},
   {WL_KMS_FORMAT_RGBA8888, GST_VIDEO_FORMAT_ABGR},
   {WL_KMS_FORMAT_BGRA8888, GST_VIDEO_FORMAT_ARGB},
--
1.9.1
